### PR TITLE
Add network report job template

### PR DESCRIPTION
**Pull request by nleiva**
The goal is to update the network report activity to gather network resource information from DevNet devices and build a report [sean style](https://github.com/network-automation/toolkit/tree/master/roles/build_report).

As of now, everything seems to work. Though, I'm using `/var/www/html` and index.html as the report filename to conveniently display the report in the `node1 web` tab of the portal (see image).

```yaml
- hosts: "{{ HOSTS }}"
  become: yes
  vars:
    report_server: node1
    #web_path: /var/www/html/reports/
    web_path: /var/www/html
```

image

The report creation could fail in the future if network resource modules make changes to the data structures or each device model deviate too much from each other.

### CODE REVIEW

File: network/report.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+# Re-write of: 
+# https://github.com/network-automation/toolkit/blob/master/playbooks/network_report.yml
+# https://github.com/network-automation/toolkit/blob/master/roles/build_report/tasks/main.yml
+
+- name: Collect facts
+  hosts: routers
```
Comment (willtome) [inline network/report.yml]: Set the hosts line to "{{ HOSTS }}" such that the user can select which networking devices to collect facts from?

File: network/report.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        gather_subset: min
+        gather_network_resources: all
+      when: ansible_network_os == 'cisco.iosxr.iosxr'
+
+    # The dig lookup requires the python 'dnspython' library  
+    # - name: Resolve IP address
+    #   ansible.builtin.set_fact:
+    #     ansible_host: "{{ lookup('community.general.dig', inventory_hostname)}}"
+
+- hosts: "{{ HOSTS }}"
```
Comment (willtome) [inline network/report.yml]: This should be able to combine with the first play with the hosts line as a variable

File: network/report.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  vars:
+    report_server: node1
+    #web_path: /var/www/html/reports/
+    web_path: /var/www/html
+  
+  tasks:
+    - ansible.builtin.include_role:
+        name: demo.patching.build_report_network
+
+    - block:
```
Comment (willtome) [inline network/report.yml:47]: In the block you can reuse the report_server role to ensure apache is installed correctly followed by creating the network report then updating the landing page. Something like this?

```
- include_role:
        name: "{{ item }}"
      loop:
        - demo.patching.report_server
        - demo.patching.build_report_network
        
    - include_role:
        name: demo.patching.report_server
        tasks_from: linux_landing_page
```

File: network/setup.yml [hunk 4]
```diff
@@ -2,30 +2,25 @@
 user_message:
 
 controller_components:
-  - organizations
   - projects
+  - credentials
```
Comment (willtome) [inline network/setup.yml]: are you creating a credential?

File: network/setup.yml [hunk 5]
```diff
@@ -67,3 +69,27 @@ controller_templates:
...
+    survey_enabled: true
+    survey:
+      name: ''
+      description: ''
+      spec:
+        - question_name: Server Name or Pattern
+          type: text
+          variable: HOSTS
+          required: true
+          default: node1
```
Comment (willtome) [inline network/setup.yml]: defaults to `routers`

File: collections/ansible_collections/demo/patching/roles/build_report_network/vars/main.yml [hunk 6]
```diff
@@ -1 +1,12 @@
-file_path: /var/www/html
\ No newline at end of file
+#file_path: "{{ web_path | default('/var/www/html/reports') }}"
+file_path: "{{ web_path | default('/var/www/html') }}"
```
Comment (willtome) [inline collections/ansible_collections/demo/patching/roles/build_report_network/vars/main.yml]: file path default to `/var/www/html/reports` will match what is configured by `report_server` role

File: collections/ansible_collections/demo/patching/roles/build_report_network/tasks/main.yml [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
+  ansible.builtin.file:
+    path: "{{ file_path }}"
+    state: directory
+    mode: '0755'
+
 - name: create HTML report
   ansible.builtin.template:
     src: report.j2
-    dest: "{{ file_path }}/network.html"
+    #dest: "{{ file_path }}/network.html"
```
Comment (willtome) [inline collections/ansible_collections/demo/patching/roles/build_report_network/tasks/main.yml]: yes, this should work when you include the landing page role

File: network/setup.yml [hunk 8]
```diff
@@ -67,3 +69,27 @@ controller_templates:
             - prefix_lists
             - snmp
             - user
+
+  - name: "NETWORK / Report"
+    job_type: check
```
Comment (willtome) [inline network/setup.yml:73]: Does check mode matter? check and run will both gather facts without making changes right?

File: collections/requirements.yml [hunk 9]
```diff
@@ -23,3 +23,11 @@ collections:
     version: 1.13.0
   - name: amazon.aws
     version: 3.1.1
+  #network
```
Comment (willtome) [inline collections/requirements.yml:26]: I will add to product demos ee

**Review by willtome — CHANGES_REQUESTED** [review]
This looks great! reusing the report_server and landing page role are the missing pieces I think.

File: collections/requirements.yml (hunk 9 above)
Comment (willtome) [reply collections/requirements.yml:26]: you already did it I see

File: network/report.yml (hunk 1 above)
Comment (nleiva) [reply network/report.yml]: Good idea. I'll make it a multi-choice option in the survey as well, with the names pre-populated.

File: network/report.yml (hunk 2 above)
Comment (nleiva) [reply network/report.yml]: I'll hard code the hosts as `node1` for the second play

File: network/setup.yml (hunk 5 above)
Comment (nleiva) [reply network/setup.yml]: Yeah, I'm thinking something like:

```yaml
      spec:
        - question_name: Routers to include?
          type: multiplechoice
          variable: HOSTS
          default: routers
          required: true
          choices:
            - sandbox-iosxe-latest-1.cisco.com
            - sandbox-iosxr-1.cisco.com
            - sandbox-nxos-1.cisco.com
            - routers
```

File: network/setup.yml (hunk 4 above)
Comment (nleiva) [reply network/setup.yml]: Nope. Good catch

File: collections/ansible_collections/demo/patching/roles/build_report_network/vars/main.yml (hunk 6 above)
Comment (nleiva) [reply collections/ansible_collections/demo/patching/roles/build_report_network/vars/main.yml]: Ok. I'm going back to `/var/www/html/reports` this time around

File: collections/ansible_collections/demo/patching/roles/build_report_network/tasks/main.yml (hunk 7 above)
Comment (nleiva) [reply collections/ansible_collections/demo/patching/roles/build_report_network/tasks/main.yml]: Got it.

File: network/setup.yml (hunk 8 above)
Comment (nleiva) [reply network/setup.yml:73]: I believe so, not 100% sure though.

**Review by willtome — APPROVED** [review]

**Comment by willtome** [comment]
This is fantastic! Great report and thank you very much for the contribution.

**Pull request merged**